.github/workflows/olympixStaticAnalysis.yml

```diff
@@ -0,0 +1,51 @@
+name: Olympix Static Analysis
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths:
+      - 'src/**.sol'
+      - '.github/workflows/olympixStaticAnalysis.yml'
+
+jobs:
+  static-analysis:
+    name: Static Analysis Security Check
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Get Changed Solidity Files
+        id: changed-files
+        uses: tj-actions/changed-files@v45
```

Comment:
I see this action has various parameters to change its behaviour. 
Please make sure that not only "changed" but also "added", "deleted" files are covered by this action. Otherwise we will run into trouble

![image](https://github.com/user-attachments/assets/d93ff45d-1a05-44cd-b86c-1ba91db4ce6e)

Reply:
sure all tested. Will run on "added", "modified", "renamed" files.
The "changes" naming is more related to differences between feature and main branches